integration-tests/suites/k8s/config_reload.go

```diff
@@ -0,0 +1,165 @@
+package k8s
+
+import (
+	"encoding/json"
+	"strings"
+	"time"
+
+	"github.com/stackrox/collector/integration-tests/pkg/collector"
+	"github.com/stackrox/collector/integration-tests/pkg/log"
+
+	coreV1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	EXT_IP_ENABLE = `
+networking:
+  externalIps:
+    enable: true
+`
+
+	EXT_IP_DISABLE = `
+networking:
+  externalIps:
+    enable: false
+`
+
+	CONFIG_MAP_NAME = "collector-config"
+)
+
+type ConfigQueryResponse struct {
+	Networking struct {
+		ExternalIps struct {
+			Enable bool
+		}
+	}
+}
+
+type K8sConfigReloadTestSuite struct {
+	K8sTestSuiteBase
+}
+
+func (k *K8sConfigReloadTestSuite) SetupSuite() {
+	k.T().Cleanup(func() {
+		k.Sensor().Stop()
+		k.teardownTargetNamespace()
+	})
+
+	k.Sensor().Start()
+}
+
+func (k *K8sConfigReloadTestSuite) AfterTest(suiteName, testName string) {
+	k.StopCollector()
+	k.Executor().RemoveConfigMap(k.TestNamespace(), CONFIG_MAP_NAME)
+}
+
+func (k *K8sConfigReloadTestSuite) TestCreateConfigurationAfterStart() {
+	k.StartCollector(&collector.StartupOptions{
+		Env: map[string]string{
+			"ROX_COLLECTOR_INTROSPECTION_ENABLE": "true",
+		},
+	})
+
+	log.Info("Checking runtime configuration is not in use")
+	k.assertNoRuntimeConfig()
+
+	log.Info("Checking external IPs is enabled")
+	configMap := coreV1.ConfigMap{
+		ObjectMeta: metaV1.ObjectMeta{
+			Name:      CONFIG_MAP_NAME,
+			Namespace: k.TestNamespace(),
+		},
+		Data: map[string]string{
+			"runtime_config.yaml": EXT_IP_ENABLE,
+		},
+	}
+	k.createConfigMap(&configMap)
+	k.AssertExternalIps(true)
+
+	log.Info("Checking external IPs is disabled")
+	configMap.Data["runtime_config.yaml"] = EXT_IP_DISABLE
+	k.updateConfigMap(&configMap)
+	k.AssertExternalIps(false)
+
+	log.Info("Checking runtime configuration is not in use")
+	k.deleteConfigMap(CONFIG_MAP_NAME)
+	k.assertNoRuntimeConfig()
```

Comment:
Would be good to create it one more time and make sure that the ConfigMap can be read after being deleted.

Reply:
If the tests started with no configuration file and it was created afterwards, why would that be different to re-adding it here?

Reply:
It is certainly possible that  a bug could lead to incorrect behavior in that case. E.g. collector might be listening to file creation events on startup, but not after the file is deleted.

Reply:
Jokes on me, there was a bug with this case.